opentelemetry-api/src/opentelemetry/attributes/__init__.py

```diff
@@ -25,23 +25,33 @@
 _logger = logging.getLogger(__name__)
 
 
-def _is_valid_attribute_value(value: types.AttributeValue) -> bool:
+def _clean_attribute_value(
+    value: types.AttributeValue, max_length: Optional[int]
+) -> Tuple[bool, Optional[types.AttributeValue]]:
     """Checks if attribute value is valid.
 
     An attribute value is valid if it is either:
         - A primitive type: string, boolean, double precision floating
             point (IEEE 754-1985) or integer.
         - An array of primitive type values. The array MUST be homogeneous,
             i.e. it MUST NOT contain values of different types.
+
+    When the ``max_length`` argument is set, any strings values longer than the value
+    are truncated and returned back as the second value.
+    If the attribute value is not modified, ``None`` is returned as the second return value.
```

Comment:
Can you also include that mutablesequences are made immutable, and that byte strings are decoded?